payment.yaml

```diff
@@ -0,0 +1,227 @@
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /orders:
+    parameters:
+      - $ref: '#/components/parameters/RequestId'
+    post:
+      tags:
+        - Orders
+      description: create an payment order
+      summary: create an payment order
+      operationId: createOrder
+      requestBody:
+        description: payment description
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Payment'
+      responses:
+        '201':
+          description: Successfully created an order and now is processing
+        default:
+          description: Generic Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+   
+  /wallets:
+    parameters:
+      - $ref: '#/components/parameters/RequestId'
+    post:
+      tags:
+        - Wallets
+      description: create wallet
+      summary: create wallet
+      operationId: createWallet
+      requestBody:
+        description: payment description
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/WalletAddress'
+
+      responses:
+        '201':
+          description: Successfully created wallet
+        default:
+          description: Generic Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  
+  /wallets/{wallet_address}:
+    parameters:
+      - $ref: '#/components/parameters/RequestId'
+      - in: path
+        name: wallet_address
+        schema:
+          type: string
+        required: true
+        description: public address of wallet
+    get:
+      tags:
+        - Wallets
+      description: get wallet information
+      summary: get wallet information
+      operationId: getWalletInfo
+      responses:
+        '200':
+          description: Successfully returned wallet info
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Wallet'
+        default:
+          description: Generic Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+
+components:
+  parameters:
+    RequestId:
+      in: header
+      name: X-REQUEST-ID
+      required: true
+      schema:
+        type: string
+      description: >-
+        A unique id for the request. A retransmitted request will have the same
+        id
+
+  schemas:
+  
+    WalletAddress:
+      type: object
+      required:
+        - wallet_address
+        - app_id
+      properties:
+        app_id:
+          type: string
+          description: for book keeping
+        wallet_address:
+          type: string
+
+    Wallet:
+      required:
+        - wallet_address
+        - kin_balance
```

Comment:
in POST, you don't have a balance, so it cannot be required

Reply:
I'm separating it into WalletRequest object and Wallet(Response) object, to enable having different requirements per scope